### ISSUE TITLE
[Bug]: "the application aborted the connection" when serving DateTime.MinValue with DataContractJsonSerializer

### ISSUE THREAD

**Issue by qcz** [CoreWCF/CoreWCF#1500]
### Duplicate ?

- [X] I have searched issues/discussions and did not find other issues/discussions reporting this bug.

### Product version

1.6.0

### Describe expected behavior

When serving JSON with `DataContractJsonSerializer` in a BasicHttpBinding, the client does not receive anything (connection fails) if a value of a field is the default value (essenially what happens is that the field remains on its default DateTime value for some responses).

No problems with SOAP or XML.

Old .NET Framework WCF serializes this whithout any problems.
Timezone is Central Europe Standard Time on the machine.

### Describe actual behavior

The following exception is thrown on the server side:

```
System.Runtime.Serialization.SerializationException
  HResult=0x8013150C
  Message=DateTime values that are greater than DateTime.MaxValue or smaller than DateTime.MinValue when converted to UTC cannot be serialized to JSON.
  Source=System.Private.DataContractSerialization
  StackTrace:
   at System.Runtime.Serialization.Json.JsonWriterDelegator.WriteDateTimeInDefaultFormat(DateTime value)

Inner Exception 1:
ArgumentOutOfRangeException: Specified argument was out of the range of valid values. Arg_ParamName_Name

```

it is called from
```CoreWCF.WebHttp.dll!CoreWCF.Dispatcher.SingleBodyParameterMessageFormatter.SingleParameterBodyWriter.OnWriteBodyContents(System.Xml.XmlDictionaryWriter writer) Line 263```

### Which binding

BasicHttp

### security

Transport

### Which .NET version

.NET 7

### Which os platform

Windows

**Comment by mconnew** [comment]
I've looked at the implementation on .NET Framework and .NET Core, and ran a few experiments and I can't see what could be causing the difference. A DateTime instance has a Kind associated with it. The possible kinds are Unspecified, Utc, and Local. Unless it's of type Utc, the DateTime instance is converted to UTC, presuming the instance is in your local timezome. This can cause a problem if your timezone is UTC + some hours, eg CET like you're using is UTC+1. The default value for DateTime is its minimum value (DateTime.MinValue). Internally the value is tracked as a Ticks value which is duration since a particular date/time, and can't be negative. If your timezone is CET, when serializing, to convert to UTC it takes the value of the DateTime instance and subtracts 1 hours from it to get it in UTC. But if your starting time is MinValue, which has an internal Ticks value of 0, subtracting 1 hour would make it negative which isn't supported. This is the case for .NET Framework and .NET. This means you can't just be specifying DateTime as a parameter without doing something else. So then I thought maybe the issue was that you have a member attributed with `[DataMember(EmitDefaultValue = false)]` and maybe that wasn't working as expected, but testing shows that works fine.  

Basically with the CET timezone, using DateTime default value without doing anything else special won't work on .NET Framework either. Can you provide a simple minimal .NET Framework WCF service and it's equivalent CoreWCF service where the .NET Framework service works, and the CoreWCF one doesn't. There's just too many things you could be doing for me to keep test possibilities (such as you creating a DateTime instance using the UTC kind, or providing a surrogate which fixes the DateTime instance before serializing, or having a parameter type as `DateTime?` or `Nullable<DateTime>`, or many other possibilities.

**Comment by qcz** [comment]
@mconnew Thanks for the detailed explanation, and I can confirm that you are right!

Originally I've tested with these two configurations:
- A "broken" CoreWCF version of the service on a dev machine configured with a central european timezone
- And the "good" Framework version on some live servers configured with UTC.

When I was trying to create a sample Framework project which behaves the same as our live services, I came to the same conclusion as you did. Then I recalled that we have a single live server whose timezone is not UTC, and I was instantly able to reproduce the problem I was originally faced with the untouched live service itself.
